app/src/main/java/com/example/tapp/Popper.java

```diff
@@ -0,0 +1,125 @@
+import java.util.Random;
+
+public class Popper extends AppCompatActivity implements Balloon.BalloonListener {
+
+    private int numTrials = 3;
+    private int numBalloons = 10;
+    private long reactionTimes[][]; // times in nanoseconds
+    private long startTime;
+    private int trialsComplete;
+    private int balloonCount;
+    private int[] mBalloonColors = new int[3];
+    private ViewGroup mContentView;
+    private int mScreenWidth, mScreenHeight;
+
+    Button buttonStart;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_popper);
+        trialsComplete = 0;
+        balloonCount = 0;
+        reactionTimes = new long[numTrials][numBalloons];
+        buttonStart = (Button) findViewById(R.id.popper_start);
+        buttonStart.setText(String.format(getString(R.string.popper_start), trialsComplete + 1));
+        mBalloonColors[0] = Color.argb(255, 255, 0, 0);
+        mBalloonColors[1] = Color.argb(255, 0, 255, 0);
+        mBalloonColors[2] = Color.argb(255, 0, 0, 255);
+        mContentView = (ViewGroup) findViewById(R.id.balloon_view);
+
+        ViewTreeObserver viewTreeObserver = mContentView.getViewTreeObserver();
+        if (viewTreeObserver.isAlive()) {
+            viewTreeObserver.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+                @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+                @Override
+                public void onGlobalLayout() {
+                    mContentView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                    // TODO: account for device rotation
+                    mScreenWidth = mContentView.getWidth();
+                    mScreenHeight = mContentView.getHeight();
+                }
+            });
+        }
+    }
+
+    public void setStart(View v) {
+        if (trialsComplete < numTrials) {
+            balloonCount = 0;
+            buttonStart.setVisibility(View.GONE);
+            Toast.makeText(getApplicationContext(), String.format(Locale.getDefault(), "Trial %d started!", trialsComplete + 1), Toast.LENGTH_SHORT).show();
+            launchBalloon();
+        } else if (trialsComplete == numTrials) {
+            double[] averages = new double[numTrials];
+            for (int i = 0; i < numTrials; i++) {
+                double sum = 0;
+                for(int j = 0; j < numBalloons; j++) {
+                    sum += reactionTimes[i][j];
+                }
+                averages[i] = sum / numBalloons;
+            }
+            String resString = "";
+            for (int i = 0; i < numTrials; i++) {
+                resString += String.format(Locale.US, "Trial %d average: %.2f sec\n", i+1,
+                        (averages[i] / 1000000000));
+            }
+            TextView results = (android.widget.TextView) findViewById(R.id.popResults);
+            results.setText(resString);
```

Comment:
The printing looks kinda dank. (And it's partially blocked by the button.)

![dank](https://cloud.githubusercontent.com/assets/1161419/23637061/80dd7b7a-02a7-11e7-9f3a-77e2822aec68.PNG)